core/reports.go

```diff
@@ -299,12 +280,166 @@ func DecodePrincipalsReportItem(in []string) (o PrincipalsReportItem, err error)
+	if i.PrincipalName != t.PrincipalName ||
+		i.PrincipalARN != t.PrincipalARN ||
+		i.PrincipalType != t.PrincipalType ||
+		i.PrincipalTags != t.PrincipalTags ||
+		i.ServiceName != t.ServiceName ||
+		i.AccessCapability != t.AccessCapability ||
+		i.ResourceARN != t.ResourceARN {
+		return false
+	}
+	return true
+}
+
+func DecodePrincipalAccessSummaryReportItem(in []string) (o PrincipalAccessSummaryReportItem, err error) {
+	if len(in) != 8 {
+		err = &IllegalArgumentError{`in`, `invalid PrincipalAccessReportItem entry`}
+		return
+	}
+	o.AnalysisTime, err = time.Parse(time.RFC3339Nano, in[0])
+	if err != nil {
+		return
+	}
+	o.PrincipalName = in[1]
+	o.PrincipalARN = in[2]
+	o.PrincipalType = in[3]
+	o.PrincipalTags = in[4]
+	o.ServiceName = in[5]
+	o.AccessCapability = in[6]
+	o.ResourceARN = in[7]
+	return
+}
+
+type ResourceAccessSummaryReportItem struct {
+	AnalysisTime     time.Time `csv:"analysis_time" json:"analysis_time"`
+	ServiceName      string    `csv:"service_name" json:"service_name"`
+	ResourceName     string    `csv:"resource_name" json:"resource_name"`
+	ResourceARN      string    `csv:"resource_arn" json:"resource_arn"`
+	AccessCapability string    `csv:"access_capability" json:"access_capability"`
+	PrincipalType    string    `csv:"principal_type" json:"principal_type"`
+	PrincipalName    string    `csv:"principal_name" json:"principal_name"`
+	PrincipalARN     string    `csv:"principal_arn" json:"principal_arn"`
+
+	ResourceTagConfidentiality string `csv:"resource_tag_confidentiality" json:"resource_tag_confidentiality"`
+}
+
+func (i ResourceAccessSummaryReportItem) Equivalent(t ResourceAccessSummaryReportItem) bool {
+	if i.PrincipalName != t.PrincipalName ||
+		i.PrincipalARN != t.PrincipalARN ||
+		i.PrincipalType != t.PrincipalType ||
+		i.ServiceName != t.ServiceName ||
+		i.AccessCapability != t.AccessCapability ||
+		i.ResourceName != t.ResourceName ||
+		i.ResourceARN != t.ResourceARN ||
+		i.ResourceTagConfidentiality != t.ResourceTagConfidentiality {
+		return false
+	}
+	return true
+}
+
+func DecodeResourceAccessSummaryReportItem(in []string) (o ResourceAccessSummaryReportItem, err error) {
+	if len(in) != 9 {
+		err = &IllegalArgumentError{`in`, `invalid ResourceAccessReportItem entry`}
+		return
+	}
+	o.AnalysisTime, err = time.Parse(time.RFC3339Nano, in[0])
+	if err != nil {
+		return
+	}
+	o.ServiceName = in[1]
+	o.ResourceName = in[2]
+	o.ResourceARN = in[3]
+	o.AccessCapability = in[4]
+	o.PrincipalType = in[5]
+	o.PrincipalName = in[6]
+	o.PrincipalARN = in[7]
+	o.ResourceTagConfidentiality = in[8]
+	return
+}
+
+func loadReport(in io.Reader, c Collector) error {
+	rr := csv.NewReader(in)
+	records, err := rr.ReadAll()
+	if err != nil {
+		return err
+	}
+
+	for i, v := range records {
+		// skip the header row
+		if i == 0 {
+			continue
+		}
+		if err := c.Collect(v); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+type Collector interface {
+	Collect(in []string) error
```

Comment:
would help to document this `Collect` method as:

> Collect one record into item storage.  Represent each record field as an array element.

Reply:
Commented the interface and the various implementing structs.